File: TEK3/DOTNET/DOT_AREA_2017/Webserver/Modules/action/CheckOnline.cs
using System;
using System.Collections.Generic;
using System.Net;

namespace Modules.action.twitch
{
    class CheckOnline : Action
    {
        WebClient client = new WebClient();
        Dictionary<string, string> fields = new Dictionary<string, string>();
        Reaction r = null;
        string content = null;

        public List<string> getFields()
        {
            return (new List<string>() { { "streamer" } });
        }

        public void setField(Dictionary<string, string> fields)
        {
            this.fields = fields;
        }

        public void addReaction(Reaction r)
        {
            this.r = r;
        }
        public void update()
        {
            if (this.r == null)
                return;

            try
            {
                content = (client.DownloadString("https://tmi.twitch.tv/group/user/" + this.fields["streamer"]));
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex);
            }

            string number = "";
            string toFind = "\"chatter_count\":";
            for (int i = 0; i < content.Length; i++)
            {
                if (i + toFind.Length <= content.Length && content.IndexOf(toFind, i, toFind.Length) == i)
                {
                    while (content[i] != ' ')
                        i++;
                    i++;
                    while (content[i] != '\n')
                        number += content[i++];
                }
            }

            int viewers = int.Parse(number);
            string message = "";

            if (viewers != 0)
                message = this.fields["streamer"] + "is now online";
            
            if (message != "")
                r.execute(message);
        }
    }
}

File: TEK3/Java/Java_area_2017/src/main/java/com/area/modules/action/twitch/CheckOnline.java
package com.area.modules.action.twitch;

import com.area.modules.Action;
import com.area.modules.Reaction;

import java.net.URL;
import java.net.URLConnection;
import java.util.*;

public class CheckOnline implements Action {
    Map<String, String> fields = new HashMap<String, String>();
    Reaction r = null;
    URLConnection connection = null;
    String content = null;


    public List<String> getFields() {
        return (new ArrayList<String>() {{
            add("streamer");
        }});
    }

    public void setField(Map<String, String> fields) {
        this.fields = fields;
    }

    public void addReaction(Reaction r) {
        this.r = r;
    }

    public void update() {
        if (this.r == null)
            return;

        try {
            connection =  new URL("https://tmi.twitch.tv/group/user/" + this.fields.get("streamer")).openConnection();
            Scanner scanner = new Scanner(connection.getInputStream());
            scanner.useDelimiter("\\Z");
            content = scanner.next();
        }catch ( Exception ex ) {
            ex.printStackTrace();
        }

        String number = "";
        String toFind = "\"chatter_count\":";
        for (int i = 0; i < content.length(); i++) {
            if (content.regionMatches(i, toFind, 0, toFind.length())) {
                while (content.charAt(i) != ' ')
                    i++;
                i++;
                while (content.charAt(i) != '\n')
                    number += content.charAt(i++);
            }
        }
        int viewers = Integer.parseInt(number);
        if (viewers != 0)
            r.execute();
    }
}
